src/main/java/uk/gov/justice/hmpps/offenderevents/services/EventRetrievalService.java

```diff
@@ -27,44 +27,51 @@
     private final PollAuditRepository repository;
     private final int pollInterval;
     private final int maxEventRangeHours;
+    private final int windBackSeconds;
 
 
     public EventRetrievalService(final ExternalApiService externalApiService,
                                  final SnsService snsService,
                                  final PollAuditRepository repository,
                                  @Value("${application.events.poll.frequency:60000}") final int pollInterval,
+                                 @Value("${wind.back.seconds:10}") int windBackSeconds,
                                  @Value("${application.events.max.range.hours:1}") final int maxEventRangeHours) {
         this.externalApiService = externalApiService;
         this.snsService = snsService;
         this.repository = repository;
         this.pollInterval = pollInterval;
         this.maxEventRangeHours = maxEventRangeHours;
+        this.windBackSeconds = windBackSeconds;
+        log.info("Using {} wind back seconds", windBackSeconds);
     }
 
-    public void pollEvents() {
+    public void pollEvents(final LocalDateTime currentTime) {
 
-        final var currentTime = now();
-        final var audit = repository.findById(POLL_NAME).orElse(PollAudit.builder().pollName(POLL_NAME).nextRunTime(currentTime.minus(pollInterval, MILLIS)).build());
+        final var latestPollTimeAllowed = currentTime.minusSeconds(windBackSeconds);
+        final var audit = repository.findById(POLL_NAME).orElse(PollAudit.builder().pollName(POLL_NAME).nextRunTime(latestPollTimeAllowed.minus(pollInterval, MILLIS)).build());
         repository.save(audit);
 
-        final var timeDifferenceHrs = audit.getNextRunTime().until(currentTime, ChronoUnit.HOURS);
-        final var endTime = timeDifferenceHrs > maxEventRangeHours ? audit.getNextRunTime().plusHours(maxEventRangeHours) : currentTime;
+        final var timeDifferenceHrs = audit.getNextRunTime().until(latestPollTimeAllowed, ChronoUnit.HOURS);
+        final var endTime = timeDifferenceHrs > maxEventRangeHours ? audit.getNextRunTime().plusHours(maxEventRangeHours) : latestPollTimeAllowed;
 
-        log.debug("Getting events between {} and {}", audit.getNextRunTime(), endTime);
-        final var events = externalApiService.getEvents(audit.getNextRunTime(), endTime);
-        log.debug("There are {} events {}", events.size(), events);
+        if (audit.getNextRunTime().compareTo(endTime) < 0) {  // This is just to handle if end the start time is before end due to wind bank seconds
```

Comment:
Not sure when this would ever be true? Are you thinking when this version of the code replaces the old version?

Reply:
yes